programs/lib/include/flexpret_assert.h

```diff
@@ -4,42 +4,17 @@
 #include "flexpret_io.h"
 #include <stdbool.h>
 
-/**
- * Important:
- *  To avoid the issue of having the standard library override the assert macro,
- *  we include it here and #undef the macro afterwards. This ensures that when 
- *  other files #include <stdlib.h>, it will not actually be included due to 
- *  header guards, and not override the macro.
- * 
- *  Without having the #include <stdlib.h> here, files that have
- * 
- *  #include <stdlib.h>
- *  #include <flexpret_assert.h>
- * 
- *  will work just fine while files with
- * 
- *  #include <flexpret_assert.h>
- *  #include <stdlib.h>
- * 
- *  will have its assert overridden by the standard library and get compilation
- *  errors.
- * 
- */
-#include <stdlib.h>
-
-#undef assert
-
 // NDEBUG is used by the standard library to filter out asserts, so it's a good
 // idea to use the same variable
 #ifdef NDEBUG
-#define assert(cond, reason) ((void)0)
+#define fp_assert(cond, fmt, ...) ((void)0)
```

Comment:
I think this might be problematic if you do something like `fp_assert(do_something() == true)`, if `NDEBUG` is defined `do_something` is optimized away. You want to do
`#define fp_assert(cond, fmt, ...) ((void) cond)` or something similar

Reply:
That is a very good point! Will fix asap :)